File: src/TheWorld/wwwroot/lib/_app/tripsController.js
!function(){"use strict";function r(r){var t=this;t.trips=[],t.newTrip={},t.errorMessage="",t.isBusy=!0,r.get("/api/trips").then(function(r){angular.copy(r.data,t.trips)},function(){t.errorMessage="Failed to load data: "+error}).finally(function(){t.isBusy=!1}),t.addTrip=function(){t.isBusy=!0,t.errorMessage="",r.post("/api/trips",t.newTrip).then(function(r){t.trips.push(r.data),t.newTrip={}},function(){t.errorMessage="Failed to Save data: "+error}).finally(function(){t.isBusy=!1})}}r.$inject=["$http"],angular.module("app-trips").controller("tripsController",r)}();

File: src/TheWorld/Controllers/Api/TripsController.cs
using AutoMapper;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using TheWorld.Models;

namespace TheWorld.Controllers.Api
{
    [Route("api/trips")]
    public class TripsController : Controller
    {
        private ILogger<TripsController> _logger;
        private IWorldRepository _repository;

        public TripsController(IWorldRepository repositry,
            ILogger<TripsController> logger)
        {
            _repository = repositry;

            _logger = logger;
        }

        [HttpGet("")]
        public IActionResult Get()
        {
            try { 
            var result = _repository.GetAllTrips();

            return Ok(Mapper.Map<IEnumerable<TripViewModel>>(result));
            }
            catch (Exception ex)
            {
                _logger.LogError($"Failed to gte all Trips: {ex}");

                return BadRequest($"Error occoured");
            }
        }

        [HttpPost("")]
        public async Task<IActionResult> Post([FromBody] TripViewModel theTrip)
        {
            if(ModelState.IsValid)
            {
                // Save to Database
                var newTrip = Mapper.Map<Trip>(theTrip);
                _repository.AddTrip(newTrip);

                if(await _repository.SaveChangesAsync())
                {
                    return Created($"api/trips/{theTrip.Name}", Mapper.Map<TripViewModel>(newTrip));
                } else
                {
                    return BadRequest("Failed to save changes to the database");
                }
            }

            return BadRequest(ModelState);
        }
    }
}
